dspace-server-webapp/src/test/java/org/dspace/app/rest/CollectionHarvestSettingsControllerIT.java

```diff
@@ -0,0 +1,312 @@
+            == HarvestTypeEnum.valueOf(json.getString("harvest_type")).getValue());
+        assertTrue(harvestedCollection.getOaiSource().equals(json.getString("oai_source")));
+        assertTrue(harvestedCollection.getOaiSetId().equals(json.getString("oai_set_id")));
+        assertTrue(harvestedCollection.getHarvestMetadataConfig().equals(json.getString("metadata_config_id")));
+    }
+
+    @Test
+    public void PutUnProcessableEntityIfIncorrectSettings() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+
+        JSONObject json = createHarvestSettingsJson("METADATA_ONLY", "https://dspace.mit.edu/iao/request", "col_1721.1_114174", "bc");
+
+        getClient(token).perform(
+            put("/api/core/collections/" + collection.getID() + "/harvester")
+                .contentType("application/json")
+                .content(json.toString()))
+            .andExpect(status().isUnprocessableEntity());
+    }
+
+    @Test
+    public void PutHarvestSettingsDeletedIfHarvestTypeIsNone() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+
+        JSONObject json = createHarvestSettingsJson("NONE", "", "", "dc");
+
+        getClient(token).perform(
+            put("/api/core/collections/" + collection.getID() + "/harvester")
+                .contentType("application/json")
+                .content(json.toString()))
+            .andExpect(status().isOk());
+
+        HarvestedCollection harvestedCollection = harvestedCollectionService.find(context, collection);
+
+        assertNull(harvestedCollection);
+    }
+
+    @Test
+    public void PutUnauthorizedIfNotAuthenticated() throws Exception {
+        getClient().perform(put("/api/core/collections/" + collection.getID() + "/harvester")
+            .contentType("application/json"))
+            .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void PutForbiddenIfNotEnoughpermissions() throws Exception {
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        getClient(token).perform(put("/api/core/collections/" + collection.getID() + "/harvester")
+            .contentType("application/json"))
+            .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void PutNotFoundIfNoSuchCollection() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+
+        String fakeUuid = "6c9a081e-f2e5-42cd-8cf8-338f64b0841b";
+        getClient(token).perform(put("/api/core/collections/" + fakeUuid + "/harvester")
+            .contentType("application/json"))
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    public void PutUnprocessableEntityIfHarvestTypeIncorrect() throws Exception {
+        String token = getAuthToken(admin.getEmail(), password);
+
+        JSONObject json = createHarvestSettingsJson("INCORRECT_HARVEST_TYPE", "https://dspace.mit.edu/oai/request", "col_1721.1_114174", "dc");
```

Comment:
Again, can we remove the "real" URL `https://dspace.mit.edu/oai/request` from this line?   Simply replace it with some sort of pretend URL like: `https://mydspace.edu/oai/request`   The URL here should not matter for the ITs.